tests/integration.rs

```diff
@@ -12,13 +13,16 @@ fn integration_test() {
         .nth(1)
         .expect("repo name should have format `<org>/<name>`");
 
-    let repo_dir = tempfile::tempdir()
-        .expect("couldn't create temp dir")
-        .path()
-        .join(crate_name);
+    let mut repo_dir = tempfile::tempdir().expect("couldn't create temp dir").into_path();
+    repo_dir.push(crate_name);
 
     let st = Command::new("git")
-        .args(&["clone", "--depth=1", &repo_url, repo_dir.to_str().unwrap()])
+        .args(&[
+            OsStr::new("clone"),
+            OsStr::new("--depth=1"),
+            OsStr::new(&repo_url),
+            OsStr::new(&repo_dir),
```

Comment:
I don't think we have to explicitly create `OsStr` here, because `args` takes `AsRef<OsStr>` which is implemented for `str`, `String`, `Path` and `PathBuf`. I might be wrong though as I've never used these APIs.

Reply:
Yes, we could edit it as:
```suggestion
            "clone".as_ref(),
            "--depth=1".as_ref(),
            repo_url.as_ref(),
            repo_dir.as_ref(),
```

But I find that the current edition is clearer to my intention.

Reply:
Ah, right, the slice values must still be of the same type, even with as_ref conversion :+1: